dapr/clients/__init__.py

```diff
@@ -55,14 +56,18 @@ def __init__(
                 UnaryStreamClientInterceptor or
                 StreamUnaryClientInterceptor or
                 StreamStreamClientInterceptor, optional): gRPC interceptors.
+            http_timeout_seconds (int): specify a timeout for http connections
         """
         super().__init__(address, interceptors)
         self.invocation_client = None
 
         invocation_protocol = settings.DAPR_API_METHOD_INVOCATION_PROTOCOL.upper()
 
         if invocation_protocol == 'HTTP':
-            self.invocation_client = DaprInvocationHttpClient(headers_callback=headers_callback)
+            if http_timeout_seconds is None:
+                http_timeout_seconds = settings.DAPR_HTTP_TIMEOUT_SECONDS
```

Comment:
Can the default be handles in the param in the method signature?

Reply:
Well, I initially had it that way, but in order to be able to manipulate settings for the sake of unit testing, I had to use this approach. 

TIL: python method signature defaults are set once with the class is loaded and are immutable after that.